definitions/npm/firebase_v5.x.x/flow_v0.34.x-/test_firebase.js

```diff
@@ -1,305 +1,322 @@
-    },
-  })()
+      snp.foo;
+    }
+  })();
 
 // #23
-firebase.auth().setPersistence(firebase.auth.Auth.Persistence.SESSION)
-firebase.auth().setPersistence('local')
+firebase.auth().setPersistence(firebase.auth.Auth.Persistence.SESSION);
+firebase.auth().setPersistence("local");
 
 // $ExpectError
-firebase.auth().setPersistence(firebase.auth.Auth.Persistence.FOO)
+firebase.auth().setPersistence(firebase.auth.Auth.Persistence.FOO);
 // $ExpectError
-firebase.auth().setPersistence('foo')
+firebase.auth().setPersistence("foo");
 
 firebase
   .firestore()
-  .doc('/foo/bar')
+  .doc("/foo/bar")
   .get()
   .then(snapshot => {
-    ;(snapshot.id: string)
+    (snapshot.id: string);
     // $ExpectError
-    snapshot.foo
+    snapshot.foo;
   })
-  .catch()
+  .catch();
 
 // #24
 firebase
   .firestore()
-  .doc('/foo/bar')
+  .doc("/foo/bar")
   .get()
   .then(snapshot => {
-    ;(snapshot.id: string)
+    (snapshot.id: string);
     // $ExpectError
-    snapshot.foo
+    snapshot.foo;
     // $ExpectError
-    snapshot.forEach(snapshot => {})
+    snapshot.forEach(snapshot => {});
   })
-  .catch()
+  .catch();
 
 // #25
-;(firebase.firestore().doc('/foo/bar').id: string)
+(firebase.firestore().doc("/foo/bar").id: string);
 
 // #26
 // $ExpectError
 firebase
   .firestore()
-  .doc('/foo/bar')
-  .limit(4)
+  .doc("/foo/bar")
+  .limit(4);
 
 // #27
 firebase
   .firestore()
-  .collection('/foo/bar')
+  .collection("/foo/bar")
   .get()
   .then(snapshots => {
     snapshots.forEach(snapshot => {
-      ;(snapshot.id: string)
-    })
+      (snapshot.id: string);
+    });
     // $ExpectError
-    snapshots.foo
+    snapshots.foo;
   })
-  .catch()
+  .catch();
 
 // #28
 firebase
   .firestore()
-  .collection('/foo')
+  .collection("/foo")
   .limit(4)
   .endAt({ foo: 4 })
   .endBefore({ bar: 5 })
   .startAt({ foo: 1 })
   .startAfter({ bar: 2 })
-  .orderBy(new firebase.firestore.FieldPath('foo', 'bar'), 'asc')
+  .orderBy(new firebase.firestore.FieldPath("foo", "bar"), "asc")
   .get()
   .then(snapshots => {
     snapshots.forEach(snapshot => {
-      ;(snapshot.id: string)
-    })
+      (snapshot.id: string);
+    });
     // $ExpectError
-    snapshots.foo
+    snapshots.foo;
   })
-  .catch()
+  .catch();
 
 // #29
 firebase
   .firestore()
-  .collection('/foo')
-  .get({ source: 'cache' })
+  .collection("/foo")
+  .get({ source: "cache" });
 
 // #30
 firebase
   .firestore()
   .getAll(
-    firebase.firestore().doc('col/doc1'), 
-    firebase.firestore().doc('col/doc2'));
+    firebase.firestore().doc("col/doc1"),
+    firebase.firestore().doc("col/doc2")
+  );
+
+// #31
+firebase
+  .firestore()
+  .collection("/foo")
+  .where("id", "==", "5")
+  .get({ source: "cache" });
+
+firebase
+  .firestore()
+  .collection("/foo")
+  .where("id", "array-contains", "5")
+  .get({ source: "cache" });
+
+firebase
+  .firestore()
+  .collection("/foo")
+  // $ExpectError
+  .where("id", "lte", "5")
+  .get({ source: "cache" });
```

Comment:
Special bonus points to the first person that refactors this test file using `describe/it` blocks. 😉  (Not required for this PR.)

Reply:
i'm tempted. the firebase library still needs some updating and it hasn't been straight forward to add tests. what is a well laid out test that you would use as a model?

also any advice on how to get the eslint/prettier rules to pick up? my editor keeps turing the single quotes to double quotes and i'm confused...